### PR TITLE
Prevent collisions on let(:name) and let(:method_name)

### PR DESCRIPTION
**Pull request by JonRowe**
This is #2461 for main

### CODE REVIEW

File: snippets/avoid_fixture_name_collision.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+if __FILE__ =~ /^snippets/
```
Comment (pirj) [inline snippets/avoid_fixture_name_collision.rb:1]: Just a note that we have no support for snippets in `main`, the code to run them is in `rails-6-1-dev` only.
But if CI passes on `rails-6-1-dev`, the patch should most possibly work for `main`, too.

**Review by pirj — APPROVED** [review]
Thanks!

**Review by benoittgt — APPROVED** [review]

File: snippets/avoid_fixture_name_collision.rb (hunk 1 above)
Comment (JonRowe) [reply snippets/avoid_fixture_name_collision.rb:1]: Is it easy-ish to backport that support?

Comment (pirj) [reply snippets/avoid_fixture_name_collision.rb:1]: Should be easy. Looks like cherry-picking https://github.com/rspec/rspec-rails/commit/040fecc98894de42a433d94a15d19f07434b77f8 and accepting the removed lines from `Gemfile` should do it.

**Comment by pirj** [comment]
Cherry-picked along with Changelog changes. Let's see if snippets are being run.

**Comment by pirj** [comment]
Nice
```
============= Starting snippets ===============
Running avoid_fixture_name_collision.rb
.

Finished in 0.00988 seconds (files took 0.4626 seconds to load)
1 example, 0 failures

Running use_active_record_false.rb
.

Finished in 0.00749 seconds (files took 0.49572 seconds to load)
1 example, 0 failures

============= Ending snippets ===============
```

**Pull request merged**